temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowClientCallsInterceptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import io.temporal.common.interceptors.WorkflowClientCallsInterceptor;
+import io.temporal.common.interceptors.WorkflowClientCallsInterceptorBase;
+import io.temporal.opentracing.OpenTracingOptions;
+import io.temporal.opentracing.SpanOperationType;
+
+public class OpenTracingWorkflowClientCallsInterceptor extends WorkflowClientCallsInterceptorBase {
+
+  private final OpenTracingOptions openTracingOptions;
+
+  public OpenTracingWorkflowClientCallsInterceptor(
+      WorkflowClientCallsInterceptor next, OpenTracingOptions openTracingOptions) {
+    super(next);
+    this.openTracingOptions = openTracingOptions;
+  }
+
+  @Override
+  public WorkflowStartOutput start(WorkflowStartInput input) {
+    createAndPassWorkflowStartSpan(
+        input, openTracingOptions.getSpanOperationNamePrefix(SpanOperationType.WORKFLOW_START));
+    return super.start(input);
+  }
+
+  @Override
+  public WorkflowSignalWithStartOutput signalWithStart(WorkflowSignalWithStartInput input) {
+    createAndPassWorkflowStartSpan(
+        input.getWorkflowStartInput(),
+        openTracingOptions.getSpanOperationNamePrefix(
+            SpanOperationType.WORKFLOW_START_WITH_SIGNAL));
+    return super.signalWithStart(input);
+  }
+
+  private static void createAndPassWorkflowStartSpan(
+      WorkflowStartInput input, String spanOperationNamePrefix) {
+    Tracer tracer = GlobalTracer.get();
+    Tracer.SpanBuilder workflowSpan =
+        createWorkflowStartSpanBuilder(tracer, input, spanOperationNamePrefix);
+    Span span = workflowSpan.start();
+    span.finish();
```

Comment:
don't close before start call completed

Reply:
same ^. go-sdk does it the same way too.